src/modules/neo4j/neo4j-container.ts

```diff
@@ -0,0 +1,112 @@
+import { GenericContainer, Wait } from "../..";
+import { Image, Tag } from "../../repo-tag";
+import { Host } from "../../docker-client-factory";
+import { DockerClient, ContainerName, ExecResult, Command } from "../../docker-client";
+import { BoundPorts } from "../../bound-ports";
+import { StartedTestContainer, OptionalStopOptions, StoppedTestContainer } from "../../test-container";
+import { Port } from "../../port";
+import { Id as ContainerId } from "../../container";
+import { RandomUuid } from "../../uuid";
+
+export class StartedNeo4jTestContainer implements StartedTestContainer {
+  constructor(
+    private startedContainer: StartedTestContainer,
+    private boltPort: number,
+    private httpPort: number,
+    private username: string,
+    private password: string
+  ) {}
+
+  public stop(options?: OptionalStopOptions): Promise<StoppedTestContainer> {
+    return this.startedContainer.stop(options);
+  }
+
+  public getContainerIpAddress(): Host {
+    return this.startedContainer.getContainerIpAddress();
+  }
+
+  public getMappedPort(port: Port): Port {
+    return this.startedContainer.getMappedPort(port);
+  }
+
+  public getName(): ContainerName {
+    return this.startedContainer.getName();
+  }
+
+  public getId(): ContainerId {
+    return this.startedContainer.getId();
+  }
+
+  public exec(command: Command[]): Promise<ExecResult> {
+    return this.startedContainer.exec(command);
+  }
+
+  public logs(): Promise<NodeJS.ReadableStream> {
```

Comment:
i'll take an action to create an abstract base class for StartedTestContainer which implements all these common methods (stop, getContainerIpAddress, getMappedPort, getName, getId, exec, logs, etc)

Reply:
The abstract class would definitely make sense.